solang-parser/src/lexer.rs

```diff
@@ -179,6 +179,7 @@ pub enum Token<'input> {
     Case,
     Default,
     YulArrow,
+    At,
```

Comment:
Our parser should be a strict super-set of eth solidity. '@' is not a token in Ethereum Solidity.

Also, I would like `forge fmt` to support out flavour of Solidity too.

Reply:
Having a configuration option wouldn't prevent tools like `forge fmt` to support our Syntax, but it would make it optional for them. If we don't provide this, it might prevent other projects from using our parser? I mean we don't have to do this inside this PR, it just came to my attention now.

Reply:
The parser only produces a parse tree, and that you get a parse tree does not make it valid solidity code. In fact, the parser tries to produce a parse tree for any possible input, even when the input is nonsense in some areas. This helps in situations like where in the language server when the developer is still typing code, and it is complete.

The parser has special constructs for invalid input like `function() return () {}` rather than `function() returns () {}` (note the missing `s`). We parse things  like `contract {}` without a contract name. There are more 

The parse tree needs picking over to ensure that Solidity is valid. If someone uses our parser, but does not want to support annotations, they can be rejected as well as the other language constructs which are not valid Solidity.